addons/website_sale/controllers/main.py

```diff
@@ -202,6 +205,16 @@ def _get_search_domain(self, search, category, attrib_values):
 
         return domain
 
+    def _check_category(self, category):
+        try:
+            category_id = int(category)
+            category = request.env['product.public.category'].browse(category_id).exists()
+            if not category:
+                raise NotFound()
+        except:
+            raise NotFound()
```

Comment:
So instead of raising a 500 error in case of an non-parseable category, you raise a 404? What's really the difference from the user perspective?

Reply:
Well, it is a big difference between 404 and 500 and I hope I do not need to explain it ;-) I would not have an issue to just ignore the wrong parameter, but was taking what was done by Odoo as an example what is wanted in such a case.

Reply:
@nim-odoo I agree with @wtaferner on this one. A 404 is a clean, user-friendly page.. Why should we raise a 500 in the first place? 500 responds to a server error but it really isn't, is it? 😉

Reply:
In this case you can rewrite probably each controller of website.
Sure, we should raise 404 and not 500 in case of category not found because archive meanwhile or ...

But Odoo never pretend to not display a internal error 500 to the end user when the end user play with the url.
If if want to make some experiment, he is free ;)

So for me, the only thing to merge is your other PR when you remove the _order from product_template in website module.

Reply:
@Yenthe666 so my comment:
> If it is about an improvement, one could simply make the 500 error page sexier. From an end user perspective, I don't think that a nice 404 instead of a nice 500 error would make a lot of difference.

Reply:
@nim-odoo so a PR for a cleaner 500 page would be allowed? Against which version?

Reply:
@JKE-be 
So, move on with the removal of the order in https://github.com/odoo/odoo/pull/26118, it is 5 before 12...I am waiting already way too long ;-)

About the rest...should we retarget to 11.5? 500 is nothing we ever want to have even if we blame the bingbot to come up with such an URL

Reply:
if bingbot is stupid, sorry but Odoo is not responsible ;) 

Maybe you can find the origin ? On google, you can see from where bing find this url...

Reply:
Hehe, Odoo is not responsible for the stupid URL, but responsible for what kind of error you return and 500 is stating...it is our fault...sorry, these are the rules of the internet ;-)

Reply:
Else, to move forward a bit in this very important thread, have you found from where these urls are coming ?

Reply:
Nope, probably wrongly generated links from Bingbot I guess, I am just speechless how you guys see not an issue with http code 500, but yeah, not a big deal, fixed it in my fork.

Reply:
For the user it does not really matter to see a beautiful 5xx or a 4xx (traceback with server details is a bit odd, but yeah, being transparent might not be the biggest deal) as long the user does not understand who is responsible for the failed request, but for everyone else it does and will lead to wrong perceptions or will draw attention to investigate further how to take advantage of such an error. 

Not sure about your example...the second one is just not properly styled and is returning 404 instead of 200. Do you see any 500? :wink: 

In my opinion HTTP Status Code 500 should never appear no matter what...server failed to fulfill a request. Anyway, sure I'll be there and looking forward seeing you guys and enjoy the days :beers: :tada: :notes: